main.py

```diff
@@ -77,19 +79,27 @@ def gtp(load_file: "The path to the network model files"=None,
 
 
 def bootstrap(
-    working_dir: 'tf.estimator working directory.',
-    model_save_path: 'Where to export the first bootstrapped generation'):
+        working_dir: 'tf.estimator working directory. If not set, defaults to a random tmp dir'=None,
+        model_save_path: 'Where to export the first bootstrapped generation'=None):
+    is_temp_dir = False
+    if working_dir is None:
+        working_dir = tempfile.mkdtemp()
+        is_temp_dir = True
+
     _ensure_dir_exists(working_dir)
     _ensure_dir_exists(os.path.dirname(model_save_path))
     dual_net.bootstrap(working_dir)
     dual_net.export_model(working_dir, model_save_path)
 
+    if is_temp_dir:
+        shutil.rmtree(working_dir)
+
 
 def train(
-    working_dir: 'tf.estimator working directory.',
-    chunk_dir: 'Directory where gathered training chunks are.',
-    model_save_path: 'Where to export the completed generation.',
-    generation_num: 'Which generation you are training.'=0):
+        working_dir: 'tf.estimator working directory.',
```

Comment:
Did the linter change these tabs all of a sudden?

Reply:
Well, autoformat, yeah.